libshaderc_spvc/src/spvc_private.cc

```diff
@@ -288,13 +289,46 @@ shaderc_spvc_compilation_result_t generate_hlsl_shader(
     const uint32_t* source, size_t source_len,
     shaderc_spvc_compile_options_t options,
     shaderc_spvc_compilation_result_t result) {
-  std::unique_ptr<spirv_cross::CompilerHLSL> compiler(
-      new (std::nothrow) spirv_cross::CompilerHLSL(source, source_len));
-  if (!compiler) {
-    result->messages.append(
-        "Unable to initialize SPIRV-Cross HLSL compiler.\n");
+
+  std::unique_ptr<spirv_cross::CompilerHLSL> compiler;
+
+  // spvc is under development, for now run spirv-cross compiler by default
+  // TODO (sarahM0): change the default to spvc when it's done
+  if (false) {
```

Comment:
Could you change this to be a build time selection, since this code in inaccessible without changing the source?

I am thinking something like adding an option ENABLE_SPVC_PARSER to CMakeLists.txt, which would default to OFF, but could be enabled by passing in -DENABLE_SPVC_PARSER=ON to CMake. Here in the source you could guard using 
```
#if ENABLE_SPVC_PARSER
...
#else
...
#endif
```

Reply:
SHADERC_ENABLE_SPVC would be a good example to follow. I guess that would mean the variable should be called SHADERC_ENABLE_SPVC_PARSER to be consistent.

Reply:
I made these changes in the second commit. However I'm missing something because adding -DSHADERC_ENABLE_SPVC_PARSER doesn't work.
Any ideas why?

Reply:
I will take a look, sorry I didn't respond to this before sending another round of comments, my view of the PR was stale apparently.

Reply:
Looks like you missed adding it as an option in the top level CMakeLists.txt.

See the beginning of the top-level CMakeLists.txt to how this is done for SHADERC_ENABLE_SPVC

Reply:
You're right. 
I was also missing adding target_compile_definitions(shaderc_spvc ...)